### ISSUE TITLE
👍 force_hex_color - Converts background-color:transparent  to background-color:#000000 (black)

### ISSUE THREAD

**Issue by TopCoder02** [tinymce/tinymce#9391]
@danoaky-tiny 

## 📝 Provide detailed reproduction steps (if any)
Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](fiddle.tiny.cloud) or similar.

**force_hex_color** - Converts `background-color:transparent`  to `background-color:#000000`
**force_hex_color** - Doesn't convert anything in `<style>` tags

### ✔️ Expected result
I don't know the right answer here but, maybe one of these two options will work

1.  `background-color:#0000000` (Notice the extra 0, which is the transparency layer)
2.  just leave it as `transparent` and not convert it. 

### ❌ Actual result
*What is the actual result of the above steps?*

Transparent color becomes black.

### ❓ Possible solution
*If you have ideas, you can list them here. Otherwise, you can delete this section.*

1. `background-color:#0000000` (Notice the extra 0, which is the transparency layer)
2. just leave it as `transparent` and not convert it. 

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by TheSpyder** [comment]
Which version? A replication case will help a lot. Although the option is named "force hex color", the fact that it forces the color value to hex seems to be by design.

The option was added to solve a regression between TinyMCE 5 and 6, but we couldn't revert it as that would be a breaking change. We're looking at a proper fix for TinyMCE 7, so your feedback and an example would be much appreciated.

**Comment by TopCoder02** [comment]
@TheSpyder 

It happens in both 6 and 7. 

Here's a fiddle:
https://fiddle.tiny.cloud/fVQGGgyZ0B/0

It shows both issues, the conversion to black and ignoring the style tags. 

**Source View**
![image](https://github.com/tinymce/tinymce/assets/119634613/036a691b-b983-4153-a45f-7f12a3d36198)

**WYSIYG View**
![image](https://github.com/tinymce/tinymce/assets/119634613/fed6215d-9a9d-4fd4-8b35-46a4f59fe659)

**Comment by TheSpyder** [comment]
yes we haven't implemented the change yet so `7-dev` will be the same as `6`.

**Comment by TopCoder02** [comment]
I doing this in the meantime, it only addresses the issue of transparent being converted to black.   The issue where it ignores and doesn't force HEX colors in the `style` tags, this won't address.

```
 editor.on('BeforeSetContent', function (e) {
         e.content = e.content.replace(/((color|background-color|border-color)\s*:\s*|background\s*:[^;]*?)\btransparent\b/g, '$1#00000000');
 });
```

**Comment by TheSpyder** [comment]
Confirmed fixed in the next release. #9414

This can be confirmed using the `7-dev` channel on fiddle.tiny.cloud

**Comment by TopCoder02** [comment]
@TheSpyder This changed doesn't address any of the issues reported here.

**Comment by TheSpyder** [comment]
Huh. I thought it was rolling back to the v5 implementation, but I see what you mean. Apologies.

**Comment by nazar-kalinichenko** [comment]
I can confirm the issue existing in v7.0.0. We are using @tinymce/tinymce-react.

@TopCoder02's [approach](https://github.com/tinymce/tinymce/issues/9391#issuecomment-1955130915) has helped.

**Comment by spocke** [comment]
Confirmed this as a bug and a regression so escalating internally.

**Comment by jimetevenard** [comment]
Also encoutered it.

**To summarize for people passing by** (correct me if I’m wrong)

- The bug already exists in `6.x` (if you  the `force_hex_color` option)
- Moste people will encouter this bug when switching to `7.x` (since the `force_hex_color` behavior is now enforced)

Anyway, thanks guys for your reactivity,
Tiny rocks ! 😎

**Comment by nuclearghost** [comment]
Any chance of getting this into a new release soon? It's effecting a lot of our users

**Comment by TheSpyder** [comment]
If you look at the linked PR, our fix was merged into what will become the 7.2 release. While you wait feel free to confirm it works for you with the `7-dev` channel on our CDN (https://fiddle.tiny.cloud/ has an easy channel picker)

**Comment by TopCoder02** [comment]
@TheSpyder The `7 Dev` channel,  converts RGB colors to HEX, but it doesn't convert any named colors to HEX like `red, green, etc`, so `transparent` is just ignored an stays `transparent`.

It also doesn't convert anything with an alpha mask like `rgb(255,255,0,.5)` or `rgba(255,255,0)` - Even-though modern browsers accepts both. It seems like it only does RGB (without alpha),  It also ignores `hsl(0, 100%, 50%);` Which is `red`

This might be by design, just making you aware.

**Comment by TheSpyder** [comment]
Yes, that's all by design. It be more or less back to what TinyMCE 5 did.

**Comment by TheSpyder** [comment]
This was released in TinyMCE 7.2, apologies for not updating the ticket sooner
